docs/developer/v1alpha1-compared-to-v1alpha2.md

```diff
@@ -0,0 +1,66 @@
+# Cluster API v1alpha1 compared to v1alpha2
+
+## Providers
+
+### v1alpha1
+
+Providers in v1alpha1 wrap behavior specific to an environment to create the infrastructure and bootstrap instances into
+Kubernetes nodes. Examples of environments that have integrated with Cluster API v1alpha1 include, aws, gcp, openstack,
+azure, vsphere and others. The provider vendors in Cluster API's controllers, registers their actuators with the
+controllers and runs a single manager (with the Cluster API controllers) to complete a Cluster API management cluster.
+
+### v1alpha2
+
+v1alpha2 introduces new providers. Cluster API's controllers are no longer vendored by providers. Instead, Cluster API 
+offers its own independent controllers that are responsible for the core types:
+
+* Cluster
+* Machine
+* MachineSet
+* MachineDeployment
+* [plus a few more](../../pkg/apis/cluster/v1alpha2)
+
+To complete a management cluster in v1alpha2, the Cluster API controllers need support from two other providers. A Bootstrap
+Provider and an Infrastructure provider.
+
+The Bootstrap provider is responsible for running a controller that generates data necessary to bootstrap an instance into a 
+Kubernetes node (cloud-init, bash, etc).
+
+The Infrastructure provider is responsible for generating actual infrastructure (networking, load balancers, instances, 
+etc) for a particular environment that can consume bootstrap data to turn the infrastructure into a Kubernetes cluster.
+
+This means that in order to have a complete management cluster that is ready to build clusters you will need three managers.
+
+v1alpha1 "providers" have become Infrastructure providers. Cluster API has moved from being a library to vendor to its
+own set of controllers to be used on every Cluster API management cluster. Bootstrap providers are an entirely new concept
+aimed at reducing the amount of kubeadm boilerplate that every provider reimplemented in v1alpha1.
+
+## Actuators
+
+### v1alpha1
+
+Actuators are interfaces that the Cluster API controller calls. A provider pulls in the generic Cluster API controller
+and then registers actuators to run specific infrastructure logic (calls to the provider cloud).
+ 
+### v1alpha2
+
+Actuators are not used in this version. Cluster API's controllers are no longer shared across providers and therefore
+they do not need to know about the actuator interface. Instead, providers communicate to each other via Cluster API's
+central objects, namely Machine and Cluster. When a user modifies a Machine with a reference each provider will notice
+that update and respond in some way. For example, the Bootstrap provider may attach BootstrapData to a Machine or the
```

Comment:
A bootstrap provider should only ever modify its own custom resource (to set status.bootstrapData). The Cluster API Machine Controller will pull (copy) the value from the bootstrap custom resource to the Machine.

Reply:
hah. this process also helps debugging my own mental model :D